substrate-node/pallets/pallet-tfgrid/src/lib.rs

```diff
@@ -149,10 +150,36 @@ pub mod pallet {
     pub type InterfaceOf<T> =
         Interface<<T as Config>::InterfaceName, <T as Config>::InterfaceMac, InterfaceIpsOf<T>>;
 
+    // Input type for location
+    pub type LocationInput = (
+        BoundedVec<u8, ConstU32<{ node::MAX_LOCATION_LENGTH }>>,
+        BoundedVec<u8, ConstU32<{ node::MAX_LOCATION_LENGTH }>>,
+    );
+    // Concrete type for location
+    pub type LocationOf<T> = <T as Config>::Location;
+
+    // Input type for country name
+    pub type CountryNameInput = BoundedVec<u8, ConstU32<{ node::MAX_COUNTRY_NAME_LENGTH }>>;
+    // Concrete country name type
+    pub type CountryNameOf<T> = <T as Config>::CountryName;
+
+    // Input type for city name
+    pub type CityNameInput = BoundedVec<u8, ConstU32<{ node::MAX_CITY_NAME_LENGTH }>>;
+    // Concrete city name type
+    pub type CityNameOf<T> = <T as Config>::CityName;
+
+    // Input type for resources
+    pub type ResourcesInput = Resources;
```

Comment:
not sure where this is used? The input is not validated as far as I can see?

Reply:
true, ressources still not validated ;)